tests/test-validate-app-receipt.js

```diff
@@ -68,11 +70,13 @@ describe('fxpay.validateAppReceipt()', function() {
 
     var validator = new helper.ReceiptValidator();
 
-    fxpay.validateAppReceipt(function(error, productInfo) {
-      assert.instanceOf(error, fxpay.errors.InvalidReceipt);
-      assert.typeOf(productInfo, 'object');
+    fxpay.validateAppReceipt().then(function() {
+      done(Error('unexpected success'));
+    }).catch(function(reason) {
+      assert.instanceOf(reason.error, fxpay.errors.InvalidReceipt);
+      assert.typeOf(reason.productInfo, 'object');
       done();
-    });
+    }).catch(done);
```

Comment:
Yeah :( If there is an uncaught exception in the previous catch then the exception is not reported. I don't know exactly why. I guess it's because the error is thrown from the event loop. It wasn't reported at all, not even at the end of all tests which I thought was odd.